handout/forms/form-builder/form-builder_custom_validation.md

```diff
@@ -0,0 +1,102 @@
+# FormBuilder Custom Validation
+
+As useful as the built-in validators are, it is very useful to be able to include your own.
+Angular 2 allows you to do just that, with minimal effort.
+
+Let's assume we are using the same Login Form, but now we also want to test that our password has an exclamation mark somewhere in it.
+
+_app/login-form.component.ts_
+```ts
+function hasExclamationMark (input: FormControl) {
+  const hasExclamation = input.value.indexOf('!') >= 0;
+
+  return hasExclamation ? null : { needsExclamation: true };
+}
+
+
+// ...
+this.password = new FormControl('', [
+  Validators.required,
+  hasExclamationMark
+]);
+```
+
+I've created a simple function which takes the `FormControl` instance and returns null if everything is fine.
+If the test fails, it returns an object with an arbitrarily named property.
+The property name is what will be used for the `.hasError()` test.
+
+_app/login-form.component.ts_
+```html
+<!-- ... -->
+<div [hidden]="!password.hasError('needsExclamation')">
+  Your password must have an exclamation mark!
+</div>
+<!-- ... -->
+```
+
+### Predefined Parameters
+Having a custom validator check for exclamation marks might be helpful, but what if you need to check for some other form of punctuation?
```

Comment:
some of the concepts are covered in our ES6/TypeScript section - we can't have every section targeting the most junior developers.  With the current course outline we have already covered Observables, Immutable and Dependency Injection.